pkg/gatewayserver/gatewayserver.go

```diff
@@ -649,9 +649,6 @@ func (gs *GatewayServer) startHandleUpstreamTask(conn connectionEntry) {
 }
 
 func (gs *GatewayServer) startUpdateConnStatsTask(conn connectionEntry) {
```

Comment:
Why does this check need to be removed ?

If the caching is not enabled at stack level (Redis caching specifically), `gs.statsRegistry` will be `nil` and the updates will panic.